src/Raven.Server/Documents/DocumentsStorage.cs

```diff
@@ -1709,6 +1709,10 @@ public static Tombstone TableValueToTombstone(JsonOperationContext context, ref
                 collectionName = ExtractCollectionName(context, doc.Data);
 
                 var flags = GetFlagsFromOldDocument(newFlags, doc.Flags, nonPersistentFlags);
+
+                if (doc.Flags.Contain(DocumentFlags.Artificial))
```

Comment:
How about renaming `GetFlagsFromOldDocument` to:
1. GetFlagsFromOldDocumentForPut
2. GetFlagsFromOldDocumentForDelete

With impl as follows:
```

GetFlagsFromOldDocumentForDelete()
{
    var flags = GetFlagsFromOldDocumentForPut();
    if (doc.Flags.Contain(DocumentFlags.Artificial))}
         flags |= DocumentFlags.Artificial;

    return flags;
}
```

Reply:
done